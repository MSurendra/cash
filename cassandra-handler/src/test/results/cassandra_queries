PREHOOK: query: CREATE EXTERNAL TABLE IF NOT EXISTS 
cassandra_keyspace1_standard1(key int, value string)
STORED BY 'org.apache.hadoop.hive.cassandra.CassandraStorageHandler'
WITH SERDEPROPERTIES ("cassandra.columns.mapping" = ":key,Standard1:value" , "cassandra.cf.name" = "Standard1" , "cassandra.host" = "127.0.0.1" , "cassandra.port" = "9170", "cassandra.cf.name" = "Standard1" , "cassandra.partitioner" = "org.apache.cassandra.dht.RandomPartitioner" )
TBLPROPERTIES ("cassandra.ks.name" = "Keyspace1")
PREHOOK: type: CREATETABLE
POSTHOOK: query: CREATE EXTERNAL TABLE IF NOT EXISTS 
cassandra_keyspace1_standard1(key int, value string)
STORED BY 'org.apache.hadoop.hive.cassandra.CassandraStorageHandler'
WITH SERDEPROPERTIES ("cassandra.columns.mapping" = ":key,Standard1:value" , "cassandra.cf.name" = "Standard1" , "cassandra.host" = "127.0.0.1" , "cassandra.port" = "9170", "cassandra.cf.name" = "Standard1" , "cassandra.partitioner" = "org.apache.cassandra.dht.RandomPartitioner" )
TBLPROPERTIES ("cassandra.ks.name" = "Keyspace1")
POSTHOOK: type: CREATETABLE
POSTHOOK: Output: default@cassandra_keyspace1_standard1
PREHOOK: query: describe cassandra_keyspace1_standard1
PREHOOK: type: DESCTABLE
POSTHOOK: query: describe cassandra_keyspace1_standard1
POSTHOOK: type: DESCTABLE
key	int	from deserializer
value	string	from deserializer
PREHOOK: query: EXPLAIN 
FROM src INSERT OVERWRITE TABLE cassandra_keyspace1_standard1 SELECT * WHERE (key%7)=0
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN 
FROM src INSERT OVERWRITE TABLE cassandra_keyspace1_standard1 SELECT * WHERE (key%7)=0
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME src))) (TOK_INSERT (TOK_DESTINATION (TOK_TAB (TOK_TABNAME cassandra_keyspace1_standard1))) (TOK_SELECT (TOK_SELEXPR TOK_ALLCOLREF)) (TOK_WHERE (= (% (TOK_TABLE_OR_COL key) 7) 0))))

STAGE DEPENDENCIES:
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-0
    Map Reduce
      Alias -> Map Operator Tree:
        src 
          TableScan
            alias: src
            Filter Operator
              predicate:
                  expr: ((key % 7) = 0)
                  type: boolean
              Filter Operator
                predicate:
                    expr: ((key % 7) = 0)
                    type: boolean
                Select Operator
                  expressions:
                        expr: key
                        type: string
                        expr: value
                        type: string
                  outputColumnNames: _col0, _col1
                  Select Operator
                    expressions:
                          expr: UDFToInteger(_col0)
                          type: int
                          expr: _col1
                          type: string
                    outputColumnNames: _col0, _col1
                    File Output Operator
                      compressed: false
                      GlobalTableId: 1
                      table:
                          input format: org.apache.hadoop.hive.cassandra.input.HiveCassandraStandardColumnInputFormat
                          output format: org.apache.hadoop.hive.cassandra.output.HiveCassandraOutputFormat
                          serde: org.apache.hadoop.hive.cassandra.serde.StandardColumnSerDe
                          name: default.cassandra_keyspace1_standard1


PREHOOK: query: FROM src INSERT OVERWRITE TABLE cassandra_keyspace1_standard1 SELECT * WHERE (key%7)=0
PREHOOK: type: QUERY
PREHOOK: Input: default@src
PREHOOK: Output: default@cassandra_keyspace1_standard1
POSTHOOK: query: FROM src INSERT OVERWRITE TABLE cassandra_keyspace1_standard1 SELECT * WHERE (key%7)=0
POSTHOOK: type: QUERY
POSTHOOK: Input: default@src
POSTHOOK: Output: default@cassandra_keyspace1_standard1
PREHOOK: query: EXPLAIN
select * from cassandra_keyspace1_standard1
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN
select * from cassandra_keyspace1_standard1
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME cassandra_keyspace1_standard1))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR TOK_ALLCOLREF))))

STAGE DEPENDENCIES:
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select * from cassandra_keyspace1_standard1
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_keyspace1_standard1
PREHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-32_494_7286511593311887975/-mr-10000
POSTHOOK: query: select * from cassandra_keyspace1_standard1
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_keyspace1_standard1
POSTHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-32_494_7286511593311887975/-mr-10000
252	val_252
126	val_126
119	val_119
196	val_196
266	val_266
392	val_392
413	val_413
983	val_98
224	val_224
427	val_427
357	val_35
203	val_203
483	val_483
469	val_469
779	val_77
79	val_0
289	val_28
399	val_399
490	val_490
364	val_364
273	val_273
462	val_462
315	val_315
238	val_238
308	val_308
322	val_322
189	val_189
429	val_42
133	val_133
217	val_217
448	val_448
105	val_105
336	val_336
846	val_84
280	val_280
287	val_287
406	val_406
497	val_497
378	val_378
708	val_70
455	val_455
175	val_175
168	val_168
PREHOOK: query: EXPLAIN
select value from cassandra_keyspace1_standard1
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN
select value from cassandra_keyspace1_standard1
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME cassandra_keyspace1_standard1))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR (TOK_TABLE_OR_COL value)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        cassandra_keyspace1_standard1 
          TableScan
            alias: cassandra_keyspace1_standard1
            Select Operator
              expressions:
                    expr: value
                    type: string
              outputColumnNames: _col0
              File Output Operator
                compressed: false
                GlobalTableId: 0
                table:
                    input format: org.apache.hadoop.mapred.TextInputFormat
                    output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select value from cassandra_keyspace1_standard1
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_keyspace1_standard1
PREHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-32_885_3161823916748354795/-mr-10000
POSTHOOK: query: select value from cassandra_keyspace1_standard1
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_keyspace1_standard1
POSTHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-32_885_3161823916748354795/-mr-10000
val_252
val_126
val_119
val_196
val_266
val_392
val_413
val_98
val_224
val_427
val_35
val_203
val_483
val_469
val_77
val_0
val_28
val_399
val_490
val_364
val_273
val_462
val_315
val_238
val_308
val_322
val_189
val_42
val_133
val_217
val_448
val_105
val_336
val_84
val_280
val_287
val_406
val_497
val_378
val_70
val_455
val_175
val_168
PREHOOK: query: EXPLAIN
select a.key,a.value,b.value from cassandra_keyspace1_standard1 a JOIN cassandra_keyspace1_standard1 b on a.key=b.key
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN
select a.key,a.value,b.value from cassandra_keyspace1_standard1 a JOIN cassandra_keyspace1_standard1 b on a.key=b.key
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_JOIN (TOK_TABREF (TOK_TABNAME cassandra_keyspace1_standard1) a) (TOK_TABREF (TOK_TABNAME cassandra_keyspace1_standard1) b) (= (. (TOK_TABLE_OR_COL a) key) (. (TOK_TABLE_OR_COL b) key)))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR (. (TOK_TABLE_OR_COL a) key)) (TOK_SELEXPR (. (TOK_TABLE_OR_COL a) value)) (TOK_SELEXPR (. (TOK_TABLE_OR_COL b) value)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        a 
          TableScan
            alias: a
            Reduce Output Operator
              key expressions:
                    expr: key
                    type: int
              sort order: +
              Map-reduce partition columns:
                    expr: key
                    type: int
              tag: 0
              value expressions:
                    expr: key
                    type: int
                    expr: value
                    type: string
        b 
          TableScan
            alias: b
            Reduce Output Operator
              key expressions:
                    expr: key
                    type: int
              sort order: +
              Map-reduce partition columns:
                    expr: key
                    type: int
              tag: 1
              value expressions:
                    expr: value
                    type: string
      Reduce Operator Tree:
        Join Operator
          condition map:
               Inner Join 0 to 1
          condition expressions:
            0 {VALUE._col0} {VALUE._col1}
            1 {VALUE._col1}
          handleSkewJoin: false
          outputColumnNames: _col0, _col1, _col5
          Select Operator
            expressions:
                  expr: _col0
                  type: int
                  expr: _col1
                  type: string
                  expr: _col5
                  type: string
            outputColumnNames: _col0, _col1, _col2
            File Output Operator
              compressed: false
              GlobalTableId: 0
              table:
                  input format: org.apache.hadoop.mapred.TextInputFormat
                  output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select a.key,a.value,b.value from cassandra_keyspace1_standard1 a JOIN cassandra_keyspace1_standard1 b on a.key=b.key
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_keyspace1_standard1
PREHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-37_515_951258717592195422/-mr-10000
POSTHOOK: query: select a.key,a.value,b.value from cassandra_keyspace1_standard1 a JOIN cassandra_keyspace1_standard1 b on a.key=b.key
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_keyspace1_standard1
POSTHOOK: Output: file:/tmp/edward/hive_2011-02-25_16-36-37_515_951258717592195422/-mr-10000
79	val_0	val_0
105	val_105	val_105
119	val_119	val_119
126	val_126	val_126
133	val_133	val_133
168	val_168	val_168
175	val_175	val_175
189	val_189	val_189
196	val_196	val_196
203	val_203	val_203
217	val_217	val_217
224	val_224	val_224
238	val_238	val_238
252	val_252	val_252
266	val_266	val_266
273	val_273	val_273
280	val_280	val_280
287	val_287	val_287
289	val_28	val_28
308	val_308	val_308
315	val_315	val_315
322	val_322	val_322
336	val_336	val_336
357	val_35	val_35
364	val_364	val_364
378	val_378	val_378
392	val_392	val_392
399	val_399	val_399
406	val_406	val_406
413	val_413	val_413
427	val_427	val_427
429	val_42	val_42
448	val_448	val_448
455	val_455	val_455
462	val_462	val_462
469	val_469	val_469
483	val_483	val_483
490	val_490	val_490
497	val_497	val_497
708	val_70	val_70
779	val_77	val_77
846	val_84	val_84
983	val_98	val_98
