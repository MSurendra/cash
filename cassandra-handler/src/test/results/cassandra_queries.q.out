PREHOOK: query: DROP TABLE cassandra_hive_table
PREHOOK: type: DROPTABLE
PREHOOK: Input: default@cassandra_hive_table
PREHOOK: Output: default@cassandra_hive_table
POSTHOOK: query: DROP TABLE cassandra_hive_table
POSTHOOK: type: DROPTABLE
POSTHOOK: Input: default@cassandra_hive_table
POSTHOOK: Output: default@cassandra_hive_table
PREHOOK: query: CREATE EXTERNAL TABLE
cassandra_hive_table(key int, value string)
STORED BY 'org.apache.hadoop.hive.cassandra.CassandraStorageHandler'
WITH SERDEPROPERTIES ("cassandra.columns.mapping" = ":key,value" , "cassandra.cf.name" = "Table" , "cassandra.host" = "127.0.0.1" , "cassandra.port" = "9170", "cassandra.partitioner" = "org.apache.cassandra.dht.RandomPartitioner" )
TBLPROPERTIES ("cassandra.ks.name" = "Hive", "cassandra.ks.repfactor" = "1", "cassandra.ks.strategy" = "org.apache.cassandra.locator.SimpleStrategy")
PREHOOK: type: CREATETABLE
POSTHOOK: query: CREATE EXTERNAL TABLE
cassandra_hive_table(key int, value string)
STORED BY 'org.apache.hadoop.hive.cassandra.CassandraStorageHandler'
WITH SERDEPROPERTIES ("cassandra.columns.mapping" = ":key,value" , "cassandra.cf.name" = "Table" , "cassandra.host" = "127.0.0.1" , "cassandra.port" = "9170", "cassandra.partitioner" = "org.apache.cassandra.dht.RandomPartitioner" )
TBLPROPERTIES ("cassandra.ks.name" = "Hive", "cassandra.ks.repfactor" = "1", "cassandra.ks.strategy" = "org.apache.cassandra.locator.SimpleStrategy")
POSTHOOK: type: CREATETABLE
POSTHOOK: Output: default@cassandra_hive_table
PREHOOK: query: DESCRIBE cassandra_hive_table
PREHOOK: type: DESCTABLE
POSTHOOK: query: DESCRIBE cassandra_hive_table
POSTHOOK: type: DESCTABLE
key	int	from deserializer
value	string	from deserializer
PREHOOK: query: EXPLAIN FROM src INSERT OVERWRITE TABLE cassandra_hive_table SELECT * WHERE (key%7)=0 ORDER BY key
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN FROM src INSERT OVERWRITE TABLE cassandra_hive_table SELECT * WHERE (key%7)=0 ORDER BY key
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME src))) (TOK_INSERT (TOK_DESTINATION (TOK_TAB (TOK_TABNAME cassandra_hive_table))) (TOK_SELECT (TOK_SELEXPR TOK_ALLCOLREF)) (TOK_WHERE (= (% (TOK_TABLE_OR_COL key) 7) 0)) (TOK_ORDERBY (TOK_TABSORTCOLNAMEASC (TOK_TABLE_OR_COL key)))))

STAGE DEPENDENCIES:
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-0
    Map Reduce
      Alias -> Map Operator Tree:
        src 
          TableScan
            alias: src
            Filter Operator
              predicate:
                  expr: ((key % 7) = 0)
                  type: boolean
              Filter Operator
                predicate:
                    expr: ((key % 7) = 0)
                    type: boolean
                Select Operator
                  expressions:
                        expr: key
                        type: string
                        expr: value
                        type: string
                  outputColumnNames: _col0, _col1
                  Reduce Output Operator
                    key expressions:
                          expr: _col0
                          type: string
                    sort order: +
                    tag: -1
                    value expressions:
                          expr: _col0
                          type: string
                          expr: _col1
                          type: string
      Reduce Operator Tree:
        Extract
          Select Operator
            expressions:
                  expr: UDFToInteger(_col0)
                  type: int
                  expr: _col1
                  type: string
            outputColumnNames: _col0, _col1
            File Output Operator
              compressed: false
              GlobalTableId: 1
              table:
                  input format: org.apache.hadoop.hive.cassandra.input.HiveCassandraStandardColumnInputFormat
                  output format: org.apache.hadoop.hive.cassandra.output.HiveCassandraOutputFormat
                  serde: org.apache.hadoop.hive.cassandra.serde.StandardColumnSerDe
                  name: default.cassandra_hive_table


PREHOOK: query: FROM src INSERT OVERWRITE TABLE cassandra_hive_table SELECT * WHERE (key%7)=0 ORDER BY key
PREHOOK: type: QUERY
PREHOOK: Input: default@src
PREHOOK: Output: default@cassandra_hive_table
POSTHOOK: query: FROM src INSERT OVERWRITE TABLE cassandra_hive_table SELECT * WHERE (key%7)=0 ORDER BY key
POSTHOOK: type: QUERY
POSTHOOK: Input: default@src
POSTHOOK: Output: default@cassandra_hive_table
PREHOOK: query: EXPLAIN select * from cassandra_hive_table ORDER BY key
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN select * from cassandra_hive_table ORDER BY key
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME cassandra_hive_table))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR TOK_ALLCOLREF)) (TOK_ORDERBY (TOK_TABSORTCOLNAMEASC (TOK_TABLE_OR_COL key)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        cassandra_hive_table 
          TableScan
            alias: cassandra_hive_table
            Select Operator
              expressions:
                    expr: key
                    type: int
                    expr: value
                    type: string
              outputColumnNames: _col0, _col1
              Reduce Output Operator
                key expressions:
                      expr: _col0
                      type: int
                sort order: +
                tag: -1
                value expressions:
                      expr: _col0
                      type: int
                      expr: _col1
                      type: string
      Reduce Operator Tree:
        Extract
          File Output Operator
            compressed: false
            GlobalTableId: 0
            table:
                input format: org.apache.hadoop.mapred.TextInputFormat
                output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select * from cassandra_hive_table ORDER BY key
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_hive_table
PREHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-49_790_6228158796402650468/-mr-10000
POSTHOOK: query: select * from cassandra_hive_table ORDER BY key
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_hive_table
POSTHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-49_790_6228158796402650468/-mr-10000
0	val_0
28	val_28
35	val_35
42	val_42
70	val_70
77	val_77
84	val_84
98	val_98
105	val_105
119	val_119
126	val_126
133	val_133
168	val_168
175	val_175
189	val_189
196	val_196
203	val_203
217	val_217
224	val_224
238	val_238
252	val_252
266	val_266
273	val_273
280	val_280
287	val_287
308	val_308
315	val_315
322	val_322
336	val_336
364	val_364
378	val_378
392	val_392
399	val_399
406	val_406
413	val_413
427	val_427
448	val_448
455	val_455
462	val_462
469	val_469
483	val_483
490	val_490
497	val_497
PREHOOK: query: EXPLAIN select key from cassandra_hive_table ORDER BY key
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN select key from cassandra_hive_table ORDER BY key
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME cassandra_hive_table))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR (TOK_TABLE_OR_COL key))) (TOK_ORDERBY (TOK_TABSORTCOLNAMEASC (TOK_TABLE_OR_COL key)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        cassandra_hive_table 
          TableScan
            alias: cassandra_hive_table
            Select Operator
              expressions:
                    expr: key
                    type: int
              outputColumnNames: _col0
              Reduce Output Operator
                key expressions:
                      expr: _col0
                      type: int
                sort order: +
                tag: -1
                value expressions:
                      expr: _col0
                      type: int
      Reduce Operator Tree:
        Extract
          File Output Operator
            compressed: false
            GlobalTableId: 0
            table:
                input format: org.apache.hadoop.mapred.TextInputFormat
                output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select key from cassandra_hive_table ORDER BY key
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_hive_table
PREHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-54_011_1589971599683254725/-mr-10000
POSTHOOK: query: select key from cassandra_hive_table ORDER BY key
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_hive_table
POSTHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-54_011_1589971599683254725/-mr-10000
0
28
35
42
70
77
84
98
105
119
126
133
168
175
189
196
203
217
224
238
252
266
273
280
287
308
315
322
336
364
378
392
399
406
413
427
448
455
462
469
483
490
497
PREHOOK: query: EXPLAIN select value from cassandra_hive_table ORDER BY VALUE
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN select value from cassandra_hive_table ORDER BY VALUE
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_TABREF (TOK_TABNAME cassandra_hive_table))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR (TOK_TABLE_OR_COL value))) (TOK_ORDERBY (TOK_TABSORTCOLNAMEASC (TOK_TABLE_OR_COL VALUE)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        cassandra_hive_table 
          TableScan
            alias: cassandra_hive_table
            Select Operator
              expressions:
                    expr: value
                    type: string
              outputColumnNames: _col0
              Reduce Output Operator
                key expressions:
                      expr: _col0
                      type: string
                sort order: +
                tag: -1
                value expressions:
                      expr: _col0
                      type: string
      Reduce Operator Tree:
        Extract
          File Output Operator
            compressed: false
            GlobalTableId: 0
            table:
                input format: org.apache.hadoop.mapred.TextInputFormat
                output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select value from cassandra_hive_table ORDER BY VALUE
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_hive_table
PREHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-58_594_5501704649094138090/-mr-10000
POSTHOOK: query: select value from cassandra_hive_table ORDER BY VALUE
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_hive_table
POSTHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-33-58_594_5501704649094138090/-mr-10000
val_0
val_105
val_119
val_126
val_133
val_168
val_175
val_189
val_196
val_203
val_217
val_224
val_238
val_252
val_266
val_273
val_28
val_280
val_287
val_308
val_315
val_322
val_336
val_35
val_364
val_378
val_392
val_399
val_406
val_413
val_42
val_427
val_448
val_455
val_462
val_469
val_483
val_490
val_497
val_70
val_77
val_84
val_98
PREHOOK: query: EXPLAIN select a.key,a.value,b.value from cassandra_hive_table a JOIN cassandra_hive_table b on a.key=b.key ORDER BY a.key
PREHOOK: type: QUERY
POSTHOOK: query: EXPLAIN select a.key,a.value,b.value from cassandra_hive_table a JOIN cassandra_hive_table b on a.key=b.key ORDER BY a.key
POSTHOOK: type: QUERY
ABSTRACT SYNTAX TREE:
  (TOK_QUERY (TOK_FROM (TOK_JOIN (TOK_TABREF (TOK_TABNAME cassandra_hive_table) a) (TOK_TABREF (TOK_TABNAME cassandra_hive_table) b) (= (. (TOK_TABLE_OR_COL a) key) (. (TOK_TABLE_OR_COL b) key)))) (TOK_INSERT (TOK_DESTINATION (TOK_DIR TOK_TMP_FILE)) (TOK_SELECT (TOK_SELEXPR (. (TOK_TABLE_OR_COL a) key)) (TOK_SELEXPR (. (TOK_TABLE_OR_COL a) value)) (TOK_SELEXPR (. (TOK_TABLE_OR_COL b) value))) (TOK_ORDERBY (TOK_TABSORTCOLNAMEASC (. (TOK_TABLE_OR_COL a) key)))))

STAGE DEPENDENCIES:
  Stage-1 is a root stage
  Stage-2 depends on stages: Stage-1
  Stage-0 is a root stage

STAGE PLANS:
  Stage: Stage-1
    Map Reduce
      Alias -> Map Operator Tree:
        a 
          TableScan
            alias: a
            Reduce Output Operator
              key expressions:
                    expr: key
                    type: int
              sort order: +
              Map-reduce partition columns:
                    expr: key
                    type: int
              tag: 0
              value expressions:
                    expr: key
                    type: int
                    expr: value
                    type: string
        b 
          TableScan
            alias: b
            Reduce Output Operator
              key expressions:
                    expr: key
                    type: int
              sort order: +
              Map-reduce partition columns:
                    expr: key
                    type: int
              tag: 1
              value expressions:
                    expr: value
                    type: string
      Reduce Operator Tree:
        Join Operator
          condition map:
               Inner Join 0 to 1
          condition expressions:
            0 {VALUE._col0} {VALUE._col1}
            1 {VALUE._col1}
          handleSkewJoin: false
          outputColumnNames: _col0, _col1, _col5
          Select Operator
            expressions:
                  expr: _col0
                  type: int
                  expr: _col1
                  type: string
                  expr: _col5
                  type: string
            outputColumnNames: _col0, _col1, _col2
            File Output Operator
              compressed: false
              GlobalTableId: 0
              table:
                  input format: org.apache.hadoop.mapred.SequenceFileInputFormat
                  output format: org.apache.hadoop.hive.ql.io.HiveSequenceFileOutputFormat

  Stage: Stage-2
    Map Reduce
      Alias -> Map Operator Tree:
        file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-34-02_477_6184217635822993214/-mr-10002 
            Reduce Output Operator
              key expressions:
                    expr: _col0
                    type: int
              sort order: +
              tag: -1
              value expressions:
                    expr: _col0
                    type: int
                    expr: _col1
                    type: string
                    expr: _col2
                    type: string
      Reduce Operator Tree:
        Extract
          File Output Operator
            compressed: false
            GlobalTableId: 0
            table:
                input format: org.apache.hadoop.mapred.TextInputFormat
                output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat

  Stage: Stage-0
    Fetch Operator
      limit: -1


PREHOOK: query: select a.key,a.value,b.value from cassandra_hive_table a JOIN cassandra_hive_table b on a.key=b.key ORDER BY a.key
PREHOOK: type: QUERY
PREHOOK: Input: default@cassandra_hive_table
PREHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-34-02_659_7467651987960490464/-mr-10000
POSTHOOK: query: select a.key,a.value,b.value from cassandra_hive_table a JOIN cassandra_hive_table b on a.key=b.key ORDER BY a.key
POSTHOOK: type: QUERY
POSTHOOK: Input: default@cassandra_hive_table
POSTHOOK: Output: file:/var/folders/b-/b-vBB2wVHpaVQzxVjeViaE+++TI/-Tmp-/yeweizhang/hive_2011-04-26_12-34-02_659_7467651987960490464/-mr-10000
0	val_0	val_0
28	val_28	val_28
35	val_35	val_35
42	val_42	val_42
70	val_70	val_70
77	val_77	val_77
84	val_84	val_84
98	val_98	val_98
105	val_105	val_105
119	val_119	val_119
126	val_126	val_126
133	val_133	val_133
168	val_168	val_168
175	val_175	val_175
189	val_189	val_189
196	val_196	val_196
203	val_203	val_203
217	val_217	val_217
224	val_224	val_224
238	val_238	val_238
252	val_252	val_252
266	val_266	val_266
273	val_273	val_273
280	val_280	val_280
287	val_287	val_287
308	val_308	val_308
315	val_315	val_315
322	val_322	val_322
336	val_336	val_336
364	val_364	val_364
378	val_378	val_378
392	val_392	val_392
399	val_399	val_399
406	val_406	val_406
413	val_413	val_413
427	val_427	val_427
448	val_448	val_448
455	val_455	val_455
462	val_462	val_462
469	val_469	val_469
483	val_483	val_483
490	val_490	val_490
497	val_497	val_497
